### DISCUSSION TITLE
Any way to specify alternative location to save offsets?

### DISCUSSION THREAD

**Discussion by zmcdargh** [MDAnalysis/mdanalysis#4609] [Development]
I often have to run analysis of simulations run by someone who has saved their simulation data to a directory where I only have read permissions. Then I often receive warnings of the following format:

```
/path/to/MDAnalysis/coordinates/XDR.py:202: UserWarning: Cannot write lock/offset file in same location as run_dir/simulation.xtc. Using slow offset calculation.
```

I'm not really sure what lock or offset files are, or how much of a slow down to expect from this issue. But I wondered if there is any way to get MDA to write the offset files somewhere else, e.g. the directory where the analysis is being run. Or is the only solution to copy these (often large) files to a new directory where I do have write access?

**Comment by RMeli** [comment]
Hi @zmcdargh.

> I'm not really sure what lock or offset files are, or how much of a slow down to expect from this issue.

The meaning of offsets is explained in our [documentation](https://docs.mdanalysis.org/stable/documentation_pages/coordinates/XDR.html#offsets-label). The offset file allows efficient random access. If you are using random access in your analysis, then it will have a significant impact.

> But I wondered if there is any way to get MDA to write the offset files somewhere else, e.g. the directory where the analysis is being run.

I'm don't think so. The name of the offsets file is defined here
https://github.com/MDAnalysis/mdanalysis/blob/347a0c0d31ee850d5a41bad5a49803bc9795f580/package/MDAnalysis/coordinates/XDR.py#L65-L66
and this function is called with the input filename.

As a workaround, have you tried creating a symlink (symbolic link) to the file you want to analyse from a directory where you have write access?

**Comment by orbeckst** [comment]
There is currently no good solution for the problem you describe. Symbolic links are sometimes useful but don't always work if the time stamps are sufficiently out of sync for the XTC or TRR reader to think that trajectory and offset file are out of sync.

Issue https://github.com/MDAnalysis/mdanalysis/issues/3793 describes a solution that would help but there's currently nobody working on it, as far as I know. You can always add a comment to the issue to say that this is a feature you'd like to see and that may shift it up in priority for someone to work on.

**Reply by orbeckst** [reply]
(And if you decide to add a comment to #3793, mention this discussion as #4609 so that a link between them is created.)
